packages/block-library/src/pattern/edit.js

```diff
@@ -1,46 +1,121 @@
+						slug
+					),
+				innerBlocks:
+					select( blockEditorStore ).getBlock( clientId )
+						?.innerBlocks,
+			};
+		},
+		[ slug, clientId ]
 	);
-
-	const { replaceBlocks, __unstableMarkNextChangeAsNotPersistent } =
-		useDispatch( blockEditorStore );
+	const {
+		replaceBlocks,
+		replaceInnerBlocks,
+		__unstableMarkNextChangeAsNotPersistent,
+	} = useDispatch( blockEditorStore );
 
 	// Run this effect when the component loads.
 	// This adds the Pattern's contents to the post.
-	// This change won't be saved.
-	// It will continue to pull from the pattern file unless changes are made to its respective template part.
 	useEffect( () => {
-		if ( selectedPattern?.blocks ) {
+		if ( selectedPattern?.blocks && ! innerBlocks?.length ) {
 			// We batch updates to block list settings to avoid triggering cascading renders
 			// for each container block included in a tree and optimize initial render.
 			// Since the above uses microtasks, we need to use a microtask here as well,
 			// because nested pattern blocks cannot be inserted if the parent block supports
 			// inner blocks but doesn't have blockSettings in the state.
 			window.queueMicrotask( () => {
 				__unstableMarkNextChangeAsNotPersistent();
+				if ( syncStatus === 'partial' ) {
+					replaceInnerBlocks(
+						clientId,
+						selectedPattern.blocks.map( ( block ) =>
+							cloneBlock( block )
+						)
+					);
+					return;
+				}
 				replaceBlocks( clientId, selectedPattern.blocks );
 			} );
 		}
-	}, [ clientId, selectedPattern?.blocks ] );
+	}, [
+		clientId,
+		selectedPattern.blocks,
+		replaceInnerBlocks,
+		__unstableMarkNextChangeAsNotPersistent,
+		innerBlocks,
+		syncStatus,
+		replaceBlocks,
+	] );
+
+	useEffect( () => {
+		if ( syncStatus !== 'partial' ) {
+			return;
+		}
+		const alignments = [ 'wide', 'full' ];
+		const blocks = innerBlocks;
+		if ( ! blocks || blocks.length === 0 ) {
+			return;
+		}
+		// Determine the widest setting of all the contained blocks.
+		const widestAlignment = blocks.reduce( ( accumulator, block ) => {
+			const { align } = block.attributes;
+			return alignments.indexOf( align ) >
+				alignments.indexOf( accumulator )
+				? align
+				: accumulator;
+		}, undefined );
 
-	const props = useBlockProps();
+		// Set the attribute of the Pattern block to match the widest
+		// alignment.
+		setAttributes( {
+			inheritedAlignment: widestAlignment ?? '',
+		} );
+	}, [
+		innerBlocks,
+		selectedPattern.blocks,
+		setAttributes,
+		inheritedAlignment,
+		syncStatus,
+	] );
 
-	return <div { ...props } />;
+	const blockProps = useBlockProps( {
+		className:
+			syncStatus === 'partial' &&
+			inheritedAlignment &&
+			`align${ inheritedAlignment }`,
+	} );
+
+	const innerBlocksProps = useInnerBlocksProps( blockProps, {
+		templateLock: syncStatus === 'partial' ? 'contentOnly' : false,
+	} );
+
+	if ( syncStatus !== 'partial' ) {
+		return <div { ...blockProps } />;
+	}
+
+	return (
+		<>
```

Comment:
Is this fragment necessary?

Reply:
Doesn't work without it, but haven't worked out why, will look closer at it tomorrow

Reply:
If you can get it working properly without the wrapper feel free to push a fix.

Reply:
Removing the fragment didn't make any difference for me. In what way wasn't this working?

Reply:
It seemed it was causing the block invalidation in my testing, but that was just me doing things in the wrong order, and invalidation was a separate issue. Have removed now.